include/xwidgets/xslider.hpp

```diff
@@ -236,8 +237,17 @@ namespace xw
     template <class D>
     inline void xslider<D>::set_defaults()
     {
-        this->_model_name() = "FloatSliderModel";
-        this->_view_name() = "FloatSliderView";
+        // TODO(C++17) constexpr
+        if (std::is_integral<value_type>::value)
+        {
+            this->_model_name() = "IntSliderModel";
+            this->_view_name() = "IntSliderView";
+        }
+        else if (std::is_floating_point<value_type>::value)
+        {
+            this->_model_name() = "FloatSliderModel";
+            this->_view_name() = "FloatSliderView";
+        }
```

Comment:
I'm surprised there is no `xint_slider` and `xfloat_slider` in xwidgets, I guess that's what I would have expected. But this looks fine to me too 👍🏽

Reply:
`xslider` is a template class, so no need for specific `xint_slider` or `xfloat_slider`